Client/Frontend/Browser/BrowserLongPressHandler.swift

```diff
@@ -0,0 +1,152 @@
+        longPressGesture?.delegate = self
+        longPressGesture?.minimumPressDuration *= 0.9
+        self.webView!.addGestureRecognizer(longPressGesture!)
+        var error: NSError?
+        let javaScript = String(contentsOfFile: NSBundle.mainBundle().pathForResource("LongPress", ofType: "js")!, encoding: NSUTF8StringEncoding, error: &error)!
+
+        let longPressUserScript = WKUserScript(source:javaScript, injectionTime: WKUserScriptInjectionTime.AtDocumentStart, forMainFrameOnly: false)
+        self.webView!.configuration.userContentController.addUserScript(longPressUserScript)
+    }
+
+
+    // MARK: - Gesture Recognizer Delegate Methods
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+
+    // MARK: - Long Press Gesture Handling
+    func didLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == UIGestureRecognizerState.Began {
+            //Finding actual touch location in webView
+            var touchLocation = gestureRecognizer.locationInView(webView)
+            touchLocation.x -= webView!.scrollView.contentInset.left
+            touchLocation.y -= webView!.scrollView.contentInset.top
+            touchLocation.x /= webView!.scrollView.zoomScale
+            touchLocation.y /= webView!.scrollView.zoomScale
+
+            evaluateTouchLocation(touchLocation, gestureRecognizer: gestureRecognizer)
+        }
+    }
+
+    ///Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
+
+        self.webView!.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler: { (elementsJSON: AnyObject!, error: NSError!) -> Void in
+
+            if elementsJSON == nil || (elementsJSON as? String) == nil {
+                return
+            }
+
+            let elementsStr = elementsJSON as String
+            let json = JSON(string: elementsStr)
+            let tags = json["tags"].asArray
+
+            if tags == nil || tags?.count == 0 {
+                return
+            }
+
+            let selectedTags = tags!.filter({ ($0.asString == "A" || $0.asString == "IMG") })
+            if (selectedTags.count == 0) {
+                return
+            }
+            //
+            let hrefElement = json["hrefElement"].asDictionary
+            var elements:[ElementType] = []
+            if hrefElement != nil {
+                let hrefStr: String = hrefElement!["hrefLink"]!.asString!
+                let hrefLinkURL = NSURL(string: hrefStr)
+                let linkRect = self.rectLocationInWebView(self.webView!, locationRect:CGRectFromString(hrefElement!["location"]!.asString!))
+                if self.isValidURL(hrefLinkURL) {
```

Comment:
Why only accept "http" and "https" schemes? This may be useful in the Open In New Tab case for schemes the browser can't handle, but I may still want to copy the URL for other schemes, for example. To keep this patch simpler, just omit this and accept all schemes for now.